### DISCUSSION TITLE
How can I read a smart contract's storage in a Foundry script?

### DISCUSSION THREAD

**Discussion by Renzo1** [Cyfrin/foundry-full-course-cu#877] [General]
How can I read a smart contract's storage in a Foundry script? 

I'm currently writing a Foundry script that requires reading data from a smart contract. In the context of Hardhat and Web3.js, I would achieve this using the following code:

contract.provider.getStorageAt(contractAddress, storageSlot);

Could you please guide me on how to accomplish something similar in Foundry?

**Comment by 0xusmanf** [comment]
```solidity
function load(address account, bytes32 slot) external returns (bytes32);
```
### Example
```solidity
/// contract LeetContract {
///     uint256 private leet = 1337; // slot 0
/// }

bytes32 leet = vm.load(address(leetContract), bytes32(uint256(0)));
emit log_uint(uint256(leet)); // 1337

```

You can find more about reading storage in foundry [here](https://book.getfoundry.sh/cheatcodes/load.html) and [here](https://book.getfoundry.sh/reference/forge-std/std-storage).
